shopfloor/services/delivery.py

```diff
@@ -130,7 +136,9 @@ def scan_deliver(self, barcode, picking_id=None):
                 return self._deliver_package(picking, package)
 
         if not barcode_valid:
-            product = search.product_from_scan(barcode)
+            product = search.product_from_scan(
+                barcode, use_packaging=(not allow_prepackaged_product)
+            )
             if product:
                 return self._deliver_product(picking, product)
```

Comment:
The delivered quantity must always be 1. Same for lot.
Usually, when the package contains a single unit, it is the same barcode on the product and package.

Reply:
You mean we should do the call like this?
```suggestion
                return self._deliver_product(picking, product, packaging_qty=1)
```

Reply:
yes but I would call it product_qty

Reply:
it is breaking existing tests, I'm doing something wrong probably.

Reply:
tests have to be adapted for that use case.

Reply:
This require some changes, I prefer to postpone that in another PR for later (draft PR created here: https://github.com/OCA/wms/pull/418 )